appscale/tools/agents/azure_agent.py

```diff
@@ -283,14 +283,19 @@ def describe_instances(self, parameters, pending=False):
       vm_list = compute_client.virtual_machine_scale_set_vms.list(resource_group,
                                                                   vmss.name)
       for vm in vm_list:
-        instance_ids.append(vm.name)
-      network_interface_list = network_client.network_interfaces.\
-        list_virtual_machine_scale_set_network_interfaces(resource_group,
-                                                          vmss.name)
-      for network_interface in network_interface_list:
-        for ip_config in network_interface.ip_configurations:
-          public_ips.append(ip_config.private_ip_address)
-          private_ips.append(ip_config.private_ip_address)
+        network_interface_list = network_client.network_interfaces. \
+          list_virtual_machine_scale_set_vm_network_interfaces(resource_group,
+                                                               vmss.name,
+                                                               vm.instance_id)
+        ip_config_private_ips = []
+        for network_interface in network_interface_list:
+          for ip_config in network_interface.ip_configurations:
+            ip_config_private_ips.append(ip_config.private_ip_address)
```

Comment:
if we only take the first one, we could change this to be:
```
ip_config_private_ip = None
for network_interface in network_interface_list:
  for ip_config in network_interface.ip_configurations:
    ip_config_private_ip = ip_config.private_ip_address
    break
  if ip_config_private_ip:
    break
```

right?

Reply:
Yes that makes total sense, I will change it.